### DISCUSSION TITLE
Consider adding Max Execution time in MySQL Data Source

### DISCUSSION THREAD

**Discussion by PeterZaitsev** [grafana/grafana#37005] [Data Sources]
In Dashboards it is easy to accidentally trigger query which will take too long time to complete potentially overloading host.  I would suggest adding option to set maximum execution time for queries which could b configured as part of data source. 

it can be done by setting max_execution_time  session variable

If you want it more generic you can allow to specify commands which will be ran at data source initialization which can set this or other custom variables

**Comment by narc-Ontakac2** [comment]
This should IMHO be done after #3168 on a by data source basis. At least all SQL data sources have this issue. The timeout from #3168 would provide a reasonable query lifetime. 

Postgres can set such a timeout on a by user basis, which gives a reasonable workaround. However MySQL can AFAIK not do that. It does however have a logon trigger provided by [init_connect](http://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_init_connect]) which could be used to implement such a workaround.

**Comment by gabor** [comment]
moved this to github issues, please continue the discussion in https://github.com/grafana/grafana/issues/78110
